api/v1/agent.proto

```diff
@@ -20,6 +20,13 @@ service Agent {
 	rpc UploadManifest(UploadManifestRequest) returns (SimpleStatusResponse) {}
 	rpc CancelUpload(CancelUploadRequest) returns (SimpleStatusResponse) {}
 
+	// Download and Map Endpoints
+	rpc Download(DownloadRequest) returns (DownloadResponse) {}
+	rpc CancelDownload(CancelDownloadRequest) returns (SimpleStatusResponse) {}
+	rpc Map(MapRequest) returns (SimpleStatusResponse) {}
+	rpc Pull(PullRequest) returns (SimpleStatusResponse) {}
+	rpc GetMapDiff(MapDiffRequest) returns (MapDiffResponse) {}
```

Comment:
@jwagenaar with regards to the commands; should we have some consistency with regards to these. In the examples you shared, we have `map -> pull`, we currently do an `upload` to upload data. Should it rather be `map -> download` for consistency? I feel we have introduced an additional concept `pull` - which you would imagine has a corresponding `push` command.

Reply:
Yes, I was thinking the following commands:
**map** will download manifest from Pennsieve and recreate folder structure locally
**map pull** downloads the files at the provided path (or individual file) to local machine
**map diff** shows information about which files are changed/renamed/moved/added, or deleted locally compared to manifests.
**map sync** will fetch an updated manifest from the server and update the local state to match the server side.
**map push** will push the local changes to the server.